plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -1137,20 +1132,21 @@ def main():
     if module.params['format'] == 'unspecified':
         module.params['format'] = None
 
-    auth_args = {'user': api_user}
-    if not (api_token_id and api_token_secret):
-        auth_args['password'] = api_password
-    else:
-        auth_args['token_name'] = api_token_id
-        auth_args['token_value'] = api_token_secret
-
-    try:
-        proxmox = ProxmoxAPI(api_host, verify_ssl=validate_certs, **auth_args)
-        global PVE_MAJOR_VERSION
-        version = proxmox_version(proxmox)
-        PVE_MAJOR_VERSION = 3 if version < LooseVersion('4.0') else version.version[0]
-    except Exception as e:
-        module.fail_json(msg='authorization on proxmox cluster failed with exception: %s' % e)
+    proxmox = ProxmoxAnsible(module).proxmox_api
+
+    pve_version = proxmox_version(proxmox)
+    # These features only work with Proxmox VE 4.0 and later
+    if pve_version < LooseVersion('4.0'):
+        if module.params['force']:
+            module.fail_json(msg="This feature only works with Proxmox VE version 4 and onwards: force")
+        for p in ['protection', 'skiplock']:
+            if module.params[p] is not None:
+                module.fail_json(msg="This feature only works with Proxmox VE version 4 and onwards: {0}".format(p))
+    # These features only work since Proxmox VE 6.0
+    if pve_version < LooseVersion('6.0'):
+        for p in ['ciuser', 'cipassword', 'sshkeys', 'ipconfig']:
+            if module.params[p] is not None:
+                module.fail_json(msg="This feature only works with Proxmox VE version 6 and onwards: {0}".format(p))
```

Comment:
As discussed on IRC chatroom #ansible-community call to fail_json should be replaced with warnings and the parameters be nullified.